packages/react-components/react-skeleton/SPEC.md

```diff
@@ -0,0 +1,107 @@
+# @fluentui/react-skeleton Spec
+
+## Background
+
+The `Skeleton` component is a temporary animation placeholder for when a service call takes time to return data and we don't want to block rendering the rest of the UI.
+
+## Prior Art
+
+### Open UI
+
+| Library    | Component Name | Spec Link                                                                       | Notes                                                                                                                                                                  |
+| ---------- | -------------- | ------------------------------------------------------------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Ant Design | Skeleton       | [Skeleton](https://ant.design/components/skeleton/)                             | Has an `avatar`, `button`, `list`, and `togglebutton`. Has animated and non-animated versions, and shape variants for the component skeletons(rounded or rectangular). |
+| Fast       | Skeleton       | [Skeleton](https://explore.fast.design/components/fast-progress)                | Has three components: `Avatar`, `List`, and `Button`. No shape variants.                                                                                               |
+| UI Fabric  | Shimmer        | [Shimmer](https://developer.microsoft.com/en-us/fluentui#/controls/web/shimmer) | Shimmer has a way to pass in custom elements as well as the defaults. There is also a prop to change the colors of the Skeleton components.                            |
+
+### Comparison of v8 and v0
+
+The existing components are:
+
+- v8
+  - `Shimmer`
+- v0
+
+## Sample Code
+
+Basic example:
+
+```jsx
+import { Skeleton } from '@fluentui/react-skeleton';
+
+function App() {
+  return <Skeleton />;
+}
+```
+
+## Variants
+
+- There are three basic `Skeleton` shapes
+  - Line
+    - This typically represents content such as paragraphs on the page
+  - Circle
+    - This typically represents content such as an `Avatar` on the page
+  - Gap
+    - This represents a space between two pieces of content being rendered on a page
+
+### Shape
+
+The `Skeleton` is a combination of any of the three shapes in order to represent the content that will be loaded onto the screen. That is, it is a combination of lines, circles, and gaps that is a visual representation of the wireframe of a page.
+
+## API
+
+### Slots
+
+- `root` - The root alot of the `Skeleton` is the container that will contain the slots that make up a `Skeleton` and any data that the `Skeleton` will load. The default html element is a `div`.
+- `wrapper` - The wrapper slot of the `Skeleton` will contain the `SkeletonElementsGroup` and the `SkeletonGradient`. The default html element is a `div`.
```

Comment:
Is there a technical reason to have this? As it's a "wrapper" inside a "wrapper":

```html
<div class="fui-Skeleton"> <!-- 👈 a wrapper -->
  <div class="fui-Skeleton__wrapper" /> <!-- 👈 a wrapper -->
</div>
```

Reply:
This is for when the data is loaded into the `data` slot of the `Skeleton`, it allows the two parts (the `Skeleton` and the `data`) to be separated as children for one `Skeleton` element

Reply:
Can you provide an example of how to use the `data` slot?

Reply:
I've moved away from the `data` slot approach in favor of using `React.Suspense` api. The SPEC has been changed to reflect this.